lib/easy.c

```diff
@@ -132,9 +138,31 @@ static CURLcode win32_init(void)
     return CURLE_FAILED_INIT;
   }
   /* The Windows Sockets DLL is acceptable. Proceed. */
+
+  /* Init pointer to ConnectEx(), a Winsock 2 function needed */
+  /* for TCP Fast Open support */
+#ifdef HAVE_CONNECTEX
+  Curl_ConnectEx = NULL;
+  sockfd = socket(AF_INET, SOCK_STREAM, 0);
+
+  if(sockfd != CURL_SOCKET_BAD) {
```

Comment:
I am missing a call to the `Curl_verify_windows_version()` in this PR (ideally, we would have a check for the exact minimum build, 1607, but I don't think that there is code in cURL yet to perform such a check), i.e. a run-time check whether it makes sense to try to activate the TCP Fast Open support. Maybe here would be a good spot?

I imagine that we would then have to test for `Curl_ConnectEx != NULL` in at least some of those places where the code currently has only a compile-time guard around the code. (And it should be a compile-time *and* runtime guard.)